addons/stock/models/stock_move.py

```diff
@@ -1635,7 +1635,7 @@ def _update_reserved_quantity(self, need, location_id, lot_id=None, package_id=N
             owner_id = self.env['res.partner']
 
         quants = self.env['stock.quant']._get_reserve_quantity(
-            self.product_id, location_id, need, uom_id=self.product_uom,
+            self.product_id, location_id, need, uom_id=self.packaging_uom_id,
```

Comment:
```suggestion
            self.product_id, location_id, need, uom_id=self.packaging_uom_id or self.product_uom,
```
in case you change the uom on the move before the reservation